docs/Feature Documentation/Smart Objects/index.md

```diff
@@ -0,0 +1,183 @@
+
+The validation of a specific Smart Object can be triggered by using the `ns_smart_device_link::ns_smart_objects::CSmartSchema::validate` method. Internally, the validate method triggers the respective validate method for each Schema Item in the object in order to perform validation.
+
+To "unapply" modifications done by the apply step, the Schema's `ns_smart_device_link::ns_smart_objects::CSmartSchema::unapplySchema` method can be used. This reverts all enum values back to their string representations.
+
+### Schema Structure
+
+Every Schema is constructed using objects called Schema Items. Each Schema Item defines the type of a specific data structure as well as any restrictions for that structure's values.
+
+In order to create a new Schema (a new object of class `ns_smart_device_link::ns_smart_objects::CSmartSchema`), you must first must define all of the required Schema Items for this object. These Schema Items can have a recursive tree structure, and each node and leaf of that tree defines structural rules for some part of the Smart Object data structure.
+
+Schema Items are represented as class hierarchy. The base class for all Schema Items is the `ns_smart_device_link::ns_smart_objects::ISchemaItem` class. This base class defines a generic validation interface for all Schema Items.
+
+- To define special elements which always fail or suceed the validation step, there are two special Schema Items: `ns_smart_device_link::ns_smart_objects::CAlwaysTrueSchemaItem` or `ns_smart_device_link::ns_smart_objects::CAlwaysFalseSchemaItem`.
+
+- `ns_smart_device_link::ns_smart_objects::CBoolSchemaItem` is used for boolean values and has no parameters, meaning that it only verifies that the Smart Object contains an actual boolean value.
+
+- `ns_smart_device_link::ns_smart_objects::TNumberSchemaItem` is a template Schema Item that can be used for both integer and floating point values. In addition to simple type verification, it is possible to set an optional min and max value range for this item.
+
+- `ns_smart_device_link::ns_smart_objects::TEnumSchemaItem` is used to verify any custom client-defined enum. It is constructed using a list of these custom enum values.
+
+- `ns_smart_device_link::ns_smart_objects::CStringSchemaItem` is used to verify a string value. In addition to simple type verification, it is possible to set an optional min and max string length for this item.
+
+- `ns_smart_device_link::ns_smart_objects::CArraySchemaItem` provides validation for an array containing values with another Schema Item. Can be used to verify an array with optional size bounds.
```

Comment:
```suggestion
- `ns_smart_device_link::ns_smart_objects::CArraySchemaItem` provides validation for an array containing values with another Schema Item. It can be used to verify an array with optional size bounds.
```